src/views/home/index.js

```diff
@@ -4,23 +4,24 @@ import { AuthContext } from '../../providers/use-auth'
 import { Navigate } from "react-router-dom";
 import './home.css'
 import { getDog } from '../../utils/fetch-request';
+import { useFirstMount } from '../../hooks';
+import { useInitDog } from '../../hooks';
 
 export const Home = () => {
   const { user } = useContext(AuthContext)
   const [left, setLeft] = useState(null)
   const [mid, setMid] = useState(null)
   const [right, setRight] = useState(null)
+  const isFirstMount = useFirstMount()
+  const dogs = useInitDog()
 
   useEffect(() => {
-    Promise.all([getDog(), getDog(), getDog()])
-      .then(dogs => {
-        const [leftDog, midDog, rightDog] = dogs.map(dog => dog.message)
-  
-        setLeft(leftDog)
-        setMid(midDog)
-        setRight(rightDog)
-      })
-  }, [])
+    if (!isFirstMount.current) {
```

Comment:
@hoangph271 
dogs was empty on first mount because useInitDog() was not completed push new url.
Im letting useEffect run on second time to prevent image reload and also not be blank

Reply:
Btw, handle the case where `dogs` is empty...?

Reply:
When dogs empty, useEffect just doin nothing, which mean defaul src './demo-main-pic.jpg' that i set will be show

Reply:
I mean this:

```jsx
  useEffect(() => {
    if (!dogs.length) return

    if (!isFirstMount.current) {
      console.info('!')
      setLeft(dogs[0])
      setMid(dogs[1])
      setRight(dogs[2])
    }
  }, [isFirstMount, dogs])
```

Reply:
@wood-crafter

Reply:
Why do we need this?

Reply:
Btw, if we handled case dogs empty, do we still need to check isFirstMount anymore?
@hoangph271

Reply:
@wood-crafter if the `dogs` is empty, these lines:
```js
setLeft(dogs[0])
setMid(dogs[1])
setRight(dogs[2])
```
Will changes the dogs value to `undefined`, and cause another (unnecessary) render

Reply:
@wood-crafter try this out...!

```jsx
useEffect(() => {
  if (dogs.length) {
    setLeft(dogs[0])
    setMid(dogs[1])
    setRight(dogs[2])
  }
}, [isFirstMount, dogs])
```

Reply:
@hoangph271 
i think that way too, but the useEffect still run 2 times, i dont know why! And im not sure if im doing all this the right way, please point me out

Reply:
@wood-crafter how was it...?

Reply:
@hoangph271 
It having bugs that the image have reload with different url.
I tried, but not found out why yet, i will work on it tonight

Reply:
> @wood-crafter maybe finish [this one](https://github.com/wood-crafter/my-tinder/pull/32) first, easy and simple...?

Will work on both

Reply:
@hoangph271  
Please review my newest changes